tests/test_bgp_af.rb

```diff
@@ -106,6 +118,38 @@ def test_collection_not_empty
     end
   end
 
+  def config_ios_xr_dependencies(asn)
+    # These dependencies are required on ios xr
```

Comment:
I added rd-auto as we had no property for that. router-id is supported so it's fine to add this here.

Regarding rd-auto - it was added recently, but I am informed that it wasn't tested on XR. So three choices:

1. Add it here (but note when you come to puppet it will be broken)
2. Use the rd-auto property but make it work for XR (I have a US open for that). The we can use the setter here - and add it to the manifest in puppet.
3. Add it to the class as per my global bgp commit.